### PR TITLE
Allowing to unsubscribe from DynDNS

### PR DESCRIPTION
**Pull request by theo-is-taken**
## The problem

If you happen to lose your DynDNS key, there's no way to reset the domain manually, and you must ask for support on the forum

## Solution

When subscribing to the DynDNS service, you can now specify a password that will be used to eventually unsubscribe later (the user must remember the password). This can be done with `yunohost dyndns subscribe -d DOMAIN -p PASSWORD` or `yunohost domain add DOMAIN -d -p PASSWORD`

Unsubscribing can either be done with a command : `yunohost dyndns unsubscribe -d DOMAIN -p PASSWORD`, or from the [Dynette's delete page](https://dyndns.yunohost.org/delete)

## PR Status

Draft

## How to test

...

### CODE REVIEW

**Comment by theo-is-taken** [comment]
Now we can unsubscribe from the DynDNS service directly from the `domain remove` command : `yunohost domain remove DOMAIN -p PASSWORD`

**Comment by zamentur** [comment]
- [ ] `yunohost tools postinstall [-h] [-p PASSWORD] [-d DOMAIN] [--subscribe [PASSWORD]|--no-subscribe]                                [--force-password] [--force-diskspace]`
- [ ] `yunohost domain add DOMAIN [--subscribe [PASSWORD]|--no-subscribe]`
- [ ] `yunohost domain remove DOMAIN [--unsubscribe [PASSWORD]|--no-unsubscribe]`
- [ ] `yunohost domain dyndns list`
- [ ] `yunohost domain dyndns subscribe`
- [ ] `yunohost domain dyndns unsubscribe`

File: src/dyndns.py [hunk 1]
```diff
@@ -170,6 +185,65 @@ def dyndns_subscribe(operation_logger, domain=None, key=None):
...
+
+    if r.status_code == 200: # Deletion was successful
+        rm(key_file, force=True)
+        # Yunohost regen conf will add the dyndns cron job if a key exists
+        # in /etc/yunohost/dyndns
+        regen_conf(["yunohost"])
+
+        # Add some dyndns update in 2 and 4 minutes from now such that user should
+        # not have to wait 10ish minutes for the conf to propagate
+        cmd = (
```
Comment (zamentur) [inline src/dyndns.py:233]: We don't need that call to dns push

Comment (theo-is-taken) [reply src/dyndns.py]: Fixed in https://github.com/YunoHost/yunohost/pull/1472/commits/dd51adcd3f467968c2663f985832e43395d0b995

File: src/dns.py [hunk 2]
```diff
@@ -622,7 +623,17 @@ def _get_registar_settings(domain):
...
 @is_unit_operation()
-def domain_dns_push(operation_logger, domain, dry_run=False, force=False, purge=False):
+def domain_dns_push(operation_logger, domains, dry_run=False, force=False, purge=False):
+    if type(domains).__name__!="list": # If we provide only a domain as an argument
+        domains = [domains]
+    for domain in domains:
+        try:
+            domain_dns_push_unique(domain,dry_run=dry_run,force=force,purge=purge)
+        except YunohostError as e:
+            logger.error(m18n.n("domain_dns_push_failed_domain",domain=domain,error=str(e)))
```
[the remark this answers is not in the document]
Comment (theo-is-taken) [reply src/dns.py:635]: Fixed in https://github.com/YunoHost/yunohost/pull/1472/commits/ac60516638dc0fdf16cfc571ecc099a89da34797

File: src/domain.py [hunk 3]
```diff
@@ -131,14 +131,14 @@ def _get_parent_domain_of(domain):
...
     Create a custom domain
 
     Keyword argument:
         domain -- Domain name to add
         dyndns -- Subscribe to DynDNS
-
+        password -- Password used to later unsubscribe from DynDNS
     """
     from yunohost.hook import hook_callback
     from yunohost.app import app_ssowatconf
```
[the remark this answers is not in the document]
Comment (theo-is-taken) [reply src/domain.py:144]: it is already checked in dyndns_subscribe

**Comment by zamentur** [comment]
## 



## TODO
### CLI
- [ ] Unit test
- [ ] Rename TSIG public_key into tsig_shared_key
- [ ] Add a Domain settings to activate or deactivate the automatic push for dynette domain (Caution: we should think the place in web admin where this should be displayed)
- [ ] Give a way to download/replace the shared key from web admin, for example via the "registrar config" ?

### Image and install process
- [ ] Adapt the internet cube and clic install process:
   - [ ] https://github.com/labriqueinternet/install/blob/master/install_procedure.py#L36 
   - [ ] https://github.com/alexAubin/clic/blob/main/install_procedure.py#L39

### Webadmin
- [ ] Adapt webadmin postinstall and domain add
- [ ] Find a way to display unsubscribed domains and subscribed domains in webadmin ?
- [ ] Add a question about unsubscribe/no-unsubscribe in the remove domain popup
- [ ] Same for domain add

### Documentation
?

### Need an API dynette change

- [ ] Allowing to unsubscribe with the shared key
- [ ] Allowing to change the passphrase `yunohost domain dyndns changepw`
- [ ] Allowing to renew the shared key with the passphrase or the old shared-key 
   - [ ] Make `postinstall` or `domain add` with the good password renew the key
   - [ ] If the domain already exist, ask if the user own the domain and ask the passphrase
   - [ ] Make the restoration on an other server change the key 
- [ ] Get the per server domain limit from dynette API

**Pull request closed without merging**